extra/models/inception.py

```diff
@@ -0,0 +1,286 @@
+      self.Conv2d_2a_3x3,
+      self.Conv2d_2b_3x3,
+      self.maxpool1,
+
+      self.Conv2d_3b_1x1,
+      self.Conv2d_4a_3x3,
+      self.maxpool2,
+
+      self.Mixed_5b,
+      self.Mixed_5c,
+      self.Mixed_5d,
+      self.Mixed_6a,
+      self.Mixed_6b,
+      self.Mixed_6c,
+      self.Mixed_6d,
+      self.Mixed_6e,
+
+      self.Mixed_7a,
+      self.Mixed_7b,
+      self.Mixed_7c,
+      self.avgpool,
+
+      lambda y: y.reshape(x.shape[0],-1),
+      self.fc,
+    ])
+
+
+# FID Inception Variation
+
+class FidInceptionA(InceptionA):
+  def __call__(self, x:Tensor) -> Tensor:
+    outputs = [
+      self.branch1x1(x),
+      x.sequential([self.branch5x5_1, self.branch5x5_2]),
+      x.sequential([self.branch3x3dbl_1, self.branch3x3dbl_2, self.branch3x3dbl_3]),
+      self.branch_pool(x.avg_pool2d(kernel_size=(3,3), stride=1, padding=1, count_include_pad=False))
+    ]
+    return Tensor.cat(*outputs, dim=1)
+
+class FidInceptionC(InceptionC):
+  def __call__(self, x:Tensor) -> Tensor:
+    outputs = [
+      self.branch1x1(x),
+      x.sequential([self.branch7x7_1, self.branch7x7_2, self.branch7x7_3]),
+      x.sequential([self.branch7x7dbl_1, self.branch7x7dbl_2, self.branch7x7dbl_3, self.branch7x7dbl_4, self.branch7x7dbl_5]),
+      self.branch_pool(x.avg_pool2d(kernel_size=(3,3), stride=1, padding=1, count_include_pad=False))
+    ]
+    return Tensor.cat(*outputs, dim=1)
+
+class FidInceptionE1(InceptionE):
+  def __call__(self, x:Tensor) -> Tensor:
+    branch3x3 = self.branch3x3_1(x)
+    branch3x3dbl = x.sequential([self.branch3x3dbl_1, self.branch3x3dbl_2])
+    outputs = [
+      self.branch1x1(x),
+      Tensor.cat(self.branch3x3_2a(branch3x3), self.branch3x3_2b(branch3x3), dim=1),
+      Tensor.cat(self.branch3x3dbl_3a(branch3x3dbl), self.branch3x3dbl_3b(branch3x3dbl), dim=1),
+      self.branch_pool(x.avg_pool2d(kernel_size=(3,3), stride=1, padding=1, count_include_pad=False)),
+    ]
+    return Tensor.cat(*outputs, dim=1)
+
+class FidInceptionE2(InceptionE):
+  def __call__(self, x:Tensor) -> Tensor:
+    branch3x3 = self.branch3x3_1(x)
+    branch3x3dbl = x.sequential([self.branch3x3dbl_1, self.branch3x3dbl_2])
+    outputs = [
+      self.branch1x1(x),
+      Tensor.cat(self.branch3x3_2a(branch3x3), self.branch3x3_2b(branch3x3), dim=1),
+      Tensor.cat(self.branch3x3dbl_3a(branch3x3dbl), self.branch3x3dbl_3b(branch3x3dbl), dim=1),
+      self.branch_pool(x.max_pool2d(kernel_size=(3,3), stride=1, padding=1)),
+    ]
+    return Tensor.cat(*outputs, dim=1)
+
+@lru_cache()
+def default_fid():
+  return torch_load(fetch("https://github.com/mseitzer/pytorch-fid/releases/download/fid_weights/pt_inception-2015-12-05-6726825d.pth", "pt_inception-2015-12-05-6726825d.pth"))
+
+class FidInceptionV3:
+  def __init__(self):
+    inception = Inception3(cls_map={
+      "A":  FidInceptionA,
+      "C":  FidInceptionC,
+      "E1": FidInceptionE1,
+      "E2": FidInceptionE2,
+    })
+    state_dict = default_fid()
```

Comment:
can these be inside a `load from_pretrained` method? preferably nothing is realized / loaded in init

Reply:
9de2e70c37e8ad5304b13505d60be5c27f2caac2

Reply:
it should be an instance method that loads in the weight once the model has inited, like resent
`https://github.com/tinygrad/tinygrad/blob/4ad83d032eabe7e1b0933c16367ffe9eb71123d1/extra/models/resnet.py#L128`, or search other `load_from_pretrained` examples

Reply:
ok, will take some reorganization since the reference model is not architected for this

Reply:
da0cf2fe0b6b8a9eb2eab99c67f30fb025a3d887

ended up better since we are realizing less weights now